proxy_agent_shared/src/error.rs

```diff
@@ -0,0 +1,64 @@
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[cfg(windows)]
+    #[error(transparent)]
+    WindowsService(#[from] windows_service::Error),
+
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+
+    #[error(transparent)]
+    Json(#[from] serde_json::Error),
+
+    #[error("Failed to create regex with error: {0}")]
+    Regex(#[from] regex::Error),
+
+    #[error("{0}")]
+    ParseVersion(ParseVersionErrorType),
+
+    #[error("Findmnt command {0}")]
+    Findmnt(String),
```

Comment:
rename it to `CommandError` like?
` #[error("{0} command: {1}")]
    CommandError(String, String),`